scripts/pkgconfig-export-targets.cmake.in

```diff
@@ -0,0 +1,113 @@
+# ----------------------------------------------------------------------
+# Auto creation of CMake targets from pkg-config
+#
+# Copyright (C) 2017 Alexandre Pretyman. All rights reserved.
+#
+# ----------------------------------------------------------------------
+
+include(FindPkgConfig)
+
+if(NOT TARGET "@PKG_CONFIG_MODULE@")
+  pkg_check_modules(@PKG_CONFIG_MODULE@ @PKG_CONFIG_MODULE@)
+  if(@PKG_CONFIG_MODULE@_FOUND)
+    add_library("@PKG_CONFIG_MODULE@" INTERFACE IMPORTED)
+
+    # --- INTERFACE_INCLUDE_DIRECTORIES begin ---
+    string(COMPARE NOTEQUAL
+        "${@PKG_CONFIG_MODULE@_INCLUDE_DIRS}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_include_dirs
+    )
+    if(_@PKG_CONFIG_MODULE@_has_include_dirs)
+      set_target_properties("@PKG_CONFIG_MODULE@"
+          PROPERTIES
+            INTERFACE_INCLUDE_DIRECTORIES
+            "${@PKG_CONFIG_MODULE@_INCLUDE_DIRS}"
+      )
+    endif()
+    # --- INTERFACE_INCLUDE_DIRECTORIES end ---
+
+    # --- INTERFACE_LINK_LIBRARIES begin ---
+    set(_@PKG_CONFIG_MODULE@_link_libs)
+    string(COMPARE NOTEQUAL
+        "${@PKG_CONFIG_MODULE@_LDFLAGS}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_ldflags
+    )
+    if(_@PKG_CONFIG_MODULE@_has_ldflags)
+      list(APPEND
+          _@PKG_CONFIG_MODULE@_link_libs
+          ${@PKG_CONFIG_MODULE@_LDFLAGS}
+      )
+    endif()
+    message("AWP: ${_@PKG_CONFIG_MODULE@_link_libs}")
+    string(COMPARE NOTEQUAL
+        "${@PKG_CONFIG_MODULE@_LDFLAGS_OTHER}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_ldflags_other
+    )
+    if(_@PKG_CONFIG_MODULE@_has_ldflags_other)
+      list(APPEND
+          _@PKG_CONFIG_MODULE@_link_libs
+          ${@PKG_CONFIG_MODULE@_LDFLAGS_OTHER}
+      )
+    endif()
+    message("AWP: ${_@PKG_CONFIG_MODULE@_link_libs}")
+
+    # No need to treat the pkg-config module's _LIBRARY_DIRS and _LIBRARIES
+    # as they are already included in LD_FLAGS
+    string(COMPARE NOTEQUAL
+        "${_@PKG_CONFIG_MODULE@_link_libs}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_link_libs
+    )
+    if(_@PKG_CONFIG_MODULE@_has_link_libs)
+      set_target_properties("@PKG_CONFIG_MODULE@"
+          PROPERTIES
+            INTERFACE_LINK_LIBRARIES
+            "${_@PKG_CONFIG_MODULE@_link_libs}"
+      )
+    endif()
+    # --- INTERFACE_LINK_LIBRARIES begin ---
+
+    # --- INTERFACE_COMPILE_OPTIONS begin ---
+    set(_@PKG_CONFIG_MODULE@_compile_opts)
+    string(COMPARE NOTEQUAL
+        "${@PKG_CONFIG_MODULE@_CFLAGS}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_cflags)
+    if(_@PKG_CONFIG_MODULE@_has_cflags)
+      list(APPEND
+          _@PKG_CONFIG_MODULE@_compile_opts
+          ${@PKG_CONFIG_MODULE@_CFLAGS}
+      )
+    endif()
+    string(COMPARE NOTEQUAL
+        "${@PKG_CONFIG_MODULE@_CFLAGS_OTHER}"
+        ""
+        _@PKG_CONFIG_MODULE@_has_cflags_other
```

Comment:
> I did not understand what you meant

Declare the function in this module and use it. Since function has it's own scope we can use nice short names:

```cmake
function(hunter_pkgconfig_export_targets ...)
  string(COMPARE EQUAL "..." "" has_ldflags)
  set(link_libs "") # we can use pretty name here
  if(has_ldflags)
    list(APPEND link_libs ${${PKG_CONFIG_MODULE}_LDFLAGS_OTHER})
  endif()
endfunction()

hunter_pkgconfig_export_targets("@PKG_CONFIG_MODULE@")
```